Objects/listobject.c

```diff
@@ -208,6 +208,19 @@ PyList_Size(PyObject *op)
         return Py_SIZE(op);
 }
 
+static inline int
+valid_index(Py_ssize_t i, Py_ssize_t limit)
```

Comment:
Why not just define this as taking two `size_t` parameters instead of doing the casting below.  The casts then happen implicitly at all call sites.